spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/resource/DeploymentStatus.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.rest.resource;
+
+import org.springframework.util.Assert;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+/**
+ * Provides a typed enumeration of deployment statuses for Streams.
+ *
+ * @author Gunnar Hillert
+ *
+ */
+@JsonFormat(shape = JsonFormat.Shape.OBJECT)
+public enum DeploymentStatus {
```

Comment:
That is unfortunate but we already have a parallel collection of DTO objects in package `org.springframework.cloud.dataflow.rest.resource`. Unless we want to add a common suffix for those objects. Honestly, I am surprised that this has not come up earlier. I cannot suffix it with `Resource` since it is not a resource object. It is basically a direct mirror object to `DeploymentState` without having any dependencies to the Deployer.

When talking to @markpollack my main idea was to add a `DeploymentStatus` property to `StreamDefinitionResource` replacing the `status` property. Technically, it is wrong to add the `statusDescription` flatly to the `StreamDefinitionResource`. BUT, for `1.2` we did not want to break backwards-compatibility. Thus, this compromise was done but I think it is directly usable for `2.0`. Also, I had to store the Statuses somewhere, rather than using a `Utils` class, I felt to add it to the `eventually correct` spot.

That being said, I still believe that `DeploymentStatus` should eventually be its own property. To indicate that I create the class the way I did, and created it in respect to serializability as well. 

As it directly, translates to the `DeploymentState` enums, I created it as an emum as well.